libsolidity/codegen/ir/IRGenerationContext.h

```diff
@@ -113,6 +125,13 @@ class IRGenerationContext
 	std::map<VariableDeclaration const*, std::pair<u256, unsigned>> m_stateVariables;
 	MultiUseYulFunctionCollector m_functions;
 	size_t m_varCounter = 0;
+
+	/// Function definitions queued for code generation. They're the functions whose calls were
+	/// discovered by the IR generator during AST traversal.
+	/// Note that the queue gets filled in a lazy way - new definitions can be added while the
+	/// collected ones get removed and traversed.
+	/// Since we don't care about duplicates or insertion order, we can use a set rather than a real queue.
```

Comment:
Good that you mention it since it is not true ;)
The order in which the code is generated will depend on the pointer values and thus code generation will not be platform-independent. I think you can use `ASTNode::CompareByID` as the comparison operator for the set to fix this.

Reply:
Ah actually this is not true - since the code that is in the end output by `MultiUseYulFunctionCollector` will be sorted by function names.

But please document that :)

Reply:
Documented, both here and in `MultiUseYulFunctionCollector`.